pallets/rewards/src/lib.rs

```diff
@@ -138,6 +141,46 @@ pub mod pallet {
 	#[pallet::generate_store(pub(super) trait Store)]
 	pub struct Pallet<T, I = ()>(_);
 
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config<I>, I: 'static = ()>
+	where
+		BalanceOf<T, I>: MaybeSerializeDeserialize,
+	{
+		pub currency_id: T::CurrencyId,
+		pub amount: BalanceOf<T, I>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config<I>, I: 'static> Default for GenesisConfig<T, I>
+	where
+		BalanceOf<T, I>: MaybeSerializeDeserialize,
+	{
+		fn default() -> Self {
+			Self {
+				currency_id: Default::default(),
+				amount: Default::default(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config<I>, I: 'static> GenesisBuild<T, I> for GenesisConfig<T, I>
+	where
+		BalanceOf<T, I>: MaybeSerializeDeserialize,
+	{
+		fn build(&self) {
+			if !self.amount.is_zero() {
+				T::Currency::mint_into(
+					self.currency_id,
+					&T::PalletId::get().into_account_truncating(),
+					self.amount,
+				)
+				.map_err(|_| log::error!("Failed to mint ED for sovereign pallet account",))
+				.ok();
+			}
```

Comment:
Out of curiosity, what would happen if we mint zero? Is it an error?

Reply:
If the account would be below the minimum balance after minting it will error

Reply:
Then, should it be `if self.amount.is_zero() < ED {` or similar?

Reply:
> If the account would be below the minimum balance after minting it will error

AFAICS, it would actually only error if we the account has reached the [max consumer threshold pre minting](https://github.com/paritytech/substrate/blob/6fa7fe1326ecaab9921c2c3888530ad679cfbb87/frame/assets/src/functions.rs#L79). Or if we [cannot have more accounts on the chain](https://github.com/paritytech/substrate/blob😅/6fa7fe1326ecaab9921c2c3888530ad679cfbb87/frame/assets/src/functions.rs#L71-L72)

Reply:
Almost never 🤣

Reply:
> It would just return Ok(())

Then, is the `if !self.amount.is_zero() {` really needed?

Reply:
Fair point 😅 It's not.